ext/libxml/ruby_xml_schema.c

```diff
@@ -209,6 +209,29 @@ static VALUE rxml_schema_elements(VALUE self)
   return result;
 }
 
+static void collect_imported_elements(xmlSchemaImportPtr import, VALUE result)
```

Comment:
This does not match the function signature for xmlHashScanner. It should take 3 parameters.

Reply:
Otherwise looks good - thanks for working on this.

Reply:
Oh, good catch!  I had just copied the collect_imported_types function and didn't think to verify it was correct.   I've now fixed both declarations to correctly match the xmlHashScanner prototype.

However, I have run into a slightly deeper issue with how these imported elements/types are being returned as a flat hash of element or type name.   Basically, I'm working with an XML standard called SCORM and the XSD is fairly large and complex.   They have a few instances where the element names have the same local-name in different namespaces.    So, I'm going to update this method to be called `imported_ns_elements` and it will instead return a hash of namespaces containing a hash of elements in that namespace.   I will also add a new `imported_ns_types` that operates the same.   This will then allow any duplicate local-name elements or type names to be found in their corresponding namespaces.

Also, one more thing: I'd like to change the methods target_namespace, name, id, version, document to instead be based on instance variables with an attribute accessor as opposed to just methods that have to build and return the value each time they're called.   I will include that change in the next update.  Just let me know if you don't like that approach and I will back it out.

Reply:
Thanks for the update. 

Adding attribute accessors seems ok to me, but make that a separate PR since its not related to the problem you are trying to solve in this one.